---
title: "Data Exploration"
author: "Gabe Runte"
date: "4/12/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(phyloseq)
library(metagenomeSeq)
library(vegan)
library(here)
library(janitor)
library(gridExtra)
```

Here, we are going to start trying to look at the data and see where that goes. 

```{r read in files}
asvs_counts = read_tsv(here("data", "ASVs_counts.tsv"))
rownames(asvs_counts) = asvs_counts$X1
asvs_counts.renamed = asvs_counts[,2:length(asvs_counts[1,])]
rownames(asvs_counts.renamed) = rownames(asvs_counts)


asvs_taxonomy = read_tsv(here("data", "ASVs_taxonomy.tsv"))
rownames(asvs_taxonomy) = asvs_taxonomy$X1
asvs_taxonomy.renamed = asvs_taxonomy[,2:length(asvs_taxonomy[1,])]
rownames(asvs_taxonomy.renamed) = rownames(asvs_taxonomy)


Metadata = read_csv(here("data", "testsamples_df.csv"))
Metadata.renamed = Metadata[,2:length(Metadata[0,])]
rownames(Metadata.renamed)<- Metadata$SampleNo

meta_15 = Metadata %>%
  filter(Year == "2015")
meta_16 = Metadata %>%
  filter(Year == "2016")
asvs_15 = asvs_counts.renamed[,colnames(asvs_counts.renamed) %in% meta_15$SampleNo]
rownames(asvs_15) = asvs_taxonomy$X1
asvs_16 = asvs_counts.renamed[,colnames(asvs_counts.renamed) %in% meta_16$SampleNo]
rownames(asvs_16) = asvs_taxonomy$X1
tax_15 = asvs_taxonomy.renamed[rownames(asvs_taxonomy.renamed) %in% rownames(asvs_15),]
rownames(tax_15) = rownames(asvs_taxonomy)

tax_16 = asvs_taxonomy.renamed[rownames(asvs_taxonomy.renamed) %in% rownames(asvs_16),]
rownames(tax_16) = rownames(asvs_taxonomy)

```

```{r}
phy.otu = otu_table(asvs_counts.renamed, taxa_are_rows = TRUE)
phy.meta = sample_data(Metadata)
phy.taxa = tax_table(as.matrix(asvs_taxonomy.renamed))
sample_names(phy.meta) = Metadata$SampleNo
narrow.phy = phyloseq(phy.otu, phy.meta, phy.taxa)


otu.15 = otu_table(asvs_15, taxa_are_rows = TRUE)
meta.15 = sample_data(meta_15)
sample_names(meta.15)= meta.15$SampleNo
taxa.15 = tax_table(as.matrix(tax_15))
phy.15 = phyloseq(otu.15, meta.15, taxa.15)

otu.16 = otu_table(asvs_16, taxa_are_rows = TRUE)
meta.16 = sample_data(meta_16)
sample_names(meta.16)= meta.16$SampleNo
taxa.16 = tax_table(as.matrix(tax_16))
phy.16 = phyloseq(otu.16, meta.16, taxa.16)
```
Tell phyloseq what the name of the sample is in the metadata and try to get the phyloseq() object to run smoothly.

```{r load function metagenome}
make_metagenomeSeq = function(physeq) {
  require("metagenomeSeq")
  require("phyloseq")
  # Enforce orientation
  if (!taxa_are_rows(physeq)) {
    physeq <- t(physeq)}
  OTU = as(otu_table(physeq), "matrix")
  # Convert sample_data to AnnotatedDataFrame
  ADF = AnnotatedDataFrame(data.frame(sample_data(physeq)))
  # define dummy 'feature' data for OTUs, using their name Helps with
  # extraction and relating to taxonomy later on.
  TDF = AnnotatedDataFrame(data.frame(OTUname = taxa_names(physeq), row.names = taxa_names(physeq)))
  # Create the metagenomeSeq object
  MGS = newMRexperiment(counts = OTU, phenoData = ADF, featureData = TDF)
  # Trigger metagenomeSeq to calculate its Cumulative Sum scaling factor.
  MGS = cumNorm(MGS)
  return(MGS)
}
```

```{r}
meta.narrow <- make_metagenomeSeq(narrow.phy)
meta_phy.15 <- make_metagenomeSeq(phy.15)
meta_phy.16 <- make_metagenomeSeq(phy.16)
```
```{r prepare and run MDS, message=FALSE, echo=false}
meta.narrow.all<-meta.narrow
narrow.css.all <-MRcounts(meta.narrow.all, norm= TRUE, log= TRUE)
narrow.csst.all <- t(narrow.css.all)
narrow.dist.all <- vegdist(narrow.csst.all, method= "bray", binary=FALSE, diag=TRUE, upper=TRUE)
set.seed(209)
narrow.mds.all <- metaMDS(narrow.dist.all, distance = "bray", k=2, trymax = 1000)

meta.phy.15<-meta_phy.15
phy.15.css.all <-MRcounts(meta.phy.15, norm= TRUE, log= TRUE)
phy.15.css.all <- t(phy.15.css.all)
phy.15.dist.all <- vegdist(phy.15.css.all, method= "bray", binary=FALSE, diag=TRUE, upper=TRUE)
set.seed(209)
phy.15.mds.all <- metaMDS(phy.15.dist.all, distance = "bray", k=2, trymax = 1000)

meta.phy.16<-meta_phy.16
phy.16.css.all <-MRcounts(meta.phy.16, norm= TRUE, log= TRUE)
phy.16.css.all <- t(phy.16.css.all)
phy.16.dist.all <- vegdist(phy.16.css.all, method= "bray", binary=FALSE, diag=TRUE, upper=TRUE)
set.seed(209)
phy.16.mds.all <- metaMDS(phy.16.dist.all, distance = "bray", k=2, trymax = 1000)


```

```{r switch otu to rra}
#meta.narrow = transform_sample_counts(meta.narrow, function(x) x/sum(x))

```

```{r building data.scores, message=FALSE}
data.scores.all <- as.data.frame(scores(narrow.mds.all))
data.scores.all$sample <- rownames(data.scores.all)

data.scores.all$Year.Station<-narrow.phy@sam_data[["Year.Station"]]
data.scores.all$Year<-narrow.phy@sam_data[["Year"]]
data.scores.all$Location<-narrow.phy@sam_data[["LocnDescriptor"]]
data.scores.all$LocationNumber<-narrow.phy@sam_data[["LocnNumber.NtoS"]]
data.scores.all$Longitude_N <- narrow.phy@sam_data[["Long-N"]]
data.scores.all$Lattitude_W <- narrow.phy@sam_data[["Lat-W"]]
data.scores.all$Depth <- narrow.phy@sam_data[["Depth"]]
data.scores.all$Temperature <-  narrow.phy@sam_data[["Temp"]]
data.scores.all$Cond <- narrow.phy@sam_data[["Cond"]]
data.scores.all$Sal <- narrow.phy@sam_data[["Sal"]]
data.scores.all$DO_pct <- narrow.phy@sam_data[["DO_pct"]]
data.scores.all$DO_mgl <- narrow.phy@sam_data[["DO_mgL"]]
data.scores.all$Chl_a <- narrow.phy@sam_data[["chl-a"]]
data.scores.all$LiveWater <- narrow.phy@sam_data[["LiveWater"]]
data.scores.all$Lugols <- narrow.phy@sam_data[["Lugols"]]
data.scores.all$Glut <- narrow.phy@sam_data[["Glut"]]
data.scores.all$Fv_Fm <- narrow.phy@sam_data[["Fv/Fm"]]
data.scores.all$Nanodrop_conc <- narrow.phy@sam_data[["Nanodrop_conc"]]
data.scores.all$Nanodrop_260280 <- narrow.phy@sam_data[["Nanodrop_260280"]]
data.scores.all$Nanodop_260230 <- narrow.phy@sam_data[["Nanodrop_260230"]]

```

```{r}
#Location and Year
 ggplot(data.scores.all) +
  geom_point(aes(NMDS1, NMDS2,  shape= as.factor(Year),  color= as.factor(Year)), size =3) +
facet_wrap(~Location)
  
 #Temp and Year
 ggplot(data.scores.all) +
  geom_point(aes(NMDS1, NMDS2,  shape= as.factor(Year),  color= Temperature), size =3) 

  
```

```{r}
#Shallow river samples were taken at lower depths  
ggplot(data.scores.all) +
  geom_point(aes(NMDS1, NMDS2,  shape= as.factor(Location),  color= Location), size =3) +
facet_wrap(~Depth)
```

```{r}
#Drop in Depth = drop in Temp
ggplot(data.scores.all) +
  geom_point(aes(NMDS1, NMDS2,  shape= as.factor(Location),  color= Temperature), size =3) +
facet_wrap(~Depth)
```

```{r}
plot_bar(narrow.phy, fill ="class")
```



```{r}
#does temperature drive dissimilarity  
narrow.diss = phyloseq::distance(narrow.phy, method = "bray")
narrow.perm = adonis(narrow.diss~ sample_data(narrow.phy)$Depth+sample_data(narrow.phy)$Temp+sample_data(narrow.phy)$Year, data=Metadata )
narrow.perm$aov.tab
```


```{r}
#does LocnDescriptor drive dissimilarity  
narrow.diss = phyloseq::distance(narrow.phy, method = "bray")
narrow.perm = adonis(narrow.diss~ sample_data(narrow.phy)$LocnDescriptor, data=Metadata )
narrow.perm$aov.tab
```


```{r}
#LiveWater messing things up?
narrow.diss = phyloseq::distance(narrow.phy, method = "bray")
narrow.perm = adonis(narrow.diss~ sample_data(narrow.phy)$LiveWater+sample_data(narrow.phy)$Year, data=Metadata )
narrow.perm$aov.tab
```


```{r}
#Temp, Year, Depth and LocnDescriptor (seems to co=vary with depth) account for almost everything outside of LiveWater
narrow.diss = phyloseq::distance(narrow.phy, method = "bray")
narrow.perm = adonis(narrow.diss~ sample_data(narrow.phy)$Depth+sample_data(narrow.phy)$Temp+sample_data(narrow.phy)$Year+sample_data(narrow.phy)$DO_pct, data=Metadata )
narrow.perm$aov.tab
```
```

